### PR TITLE
[REVIEW] First pass of `pd.read_orc` changes in tests

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This PR changes calls going via `pyarrow` and then `to_pandas` to directly call `pd.read_orc`. How-ever since `pd.read_orc` was added in pandas 1.0, we will need to version the call to this constructor. This PR does that.

Partially contributes to #11540 
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_orc.py [hunk 1]
```diff
@@ -1696,7 +1607,7 @@ def test_orc_writer_rle_stream_size(datadir, tmpdir):
 
     # Segfaults when RLE stream sizes don't account for varint length
     pa_out = pa.orc.ORCFile(reencoded).read()
-    assert_eq(df.to_pandas(), pa_out)
+    assert df.to_arrow().equals(pa_out)
```
Comment (vuule) [inline python/cudf/cudf/tests/test_orc.py:1610]: looks like this was a different kind of redundant casting?

**Review by vuule — APPROVED** [review]
Like the change!

Can we expect noticeably lower test execution time with this change? I don't have a good sense of how efficient the conversion is(was).

File: python/cudf/cudf/tests/test_orc.py (hunk 1 above)
Comment (galipremsagar) [reply python/cudf/cudf/tests/test_orc.py:1610]: Yup. We were going to pandas for comparison previously. Now we just do arrow level comparison.

**Review by wence- — APPROVED** [review]
Thanks! My only comment is whether we need to support pandas pre 1.0 at all?

**Comment by galipremsagar** [comment]
> Thanks! My only comment is whether we need to support pandas pre 1.0 at all?

Good catch. Rapids only support pandas >=1.0, so dropped the pre 1.0 code.

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**